%YAML:1.0
---

# ****************************************************************************
# *****     注意：布尔型参数和整型参数不能有小数部分，浮点型参数必须有小数部分     *****
# ****************************************************************************

# -----     相机的标识符(字符串)     -----
IP: '192.168.10.100'

# -----     相机的运行时参数     -----
RuntimeParam:
  # 是否在线运行(布尔型)；在线运行：1，离线运行：0
  IsOnline: 1
  # 是否录制视频(布尔型)；录制视频：1，不录制视频：0
  IsRecordVideo: 0
  # 更新数据帧任务的CPU内核编号(整型)；CPU内核由操作系统调度：-1，CPU内核固定：>=0
  UpdateDataCpuCore: 0
  # 录制视频流任务的CPU内核编号(整型)；CPU内核由操作系统调度：-1，CPU内核固定：>=0
  RecordVideoCpuCore: 2
  # 录制视频的帧率(浮点型)
  RecordVideoFps: 20.0
  # 离线视频文件名称(字符串)
  OfflineVideoName: '/home/cubot/data/video.avi'
  # 录像文件存储路径(字符串)，必须以“/”结尾
  RecordVideoPath: '/home/cubot/data/'

# -----     相机的模型参数     -----
ModelParam:
  # 相机OpenCV格式的内参矩阵
  CvIntrinsics: !!opencv-matrix
    rows: 3
    cols: 3
    dt: d
    data: [1713.920744,    0.000000000,    640.7787068,
           0.000000000,    1713.879851,    523.5226304,
           0.000000000,    0.000000000,    1.000000000]

  # 相机OpenCV格式的外参矩阵
  # x轴正向：枪管方向水平向右；y轴正向：枪管方向垂直向下；z轴正向：枪管方向
  # data[0, 3]，data[1, 3]，data[2, 3]为相机沿x轴、y轴、z轴的平移参数；单位：mm
  # 相机沿各轴的正向平移，平移参数取负值；相机沿各轴的负向平移，平移参数取正值
  CvExtrinsics: !!opencv-matrix
    rows: 4
    cols: 4
    dt: d
    data: [1,   0,    0,    0,
           0,   1,    0,    -50.3,
           0,   0,    1,    153.61,
           0,   0,    0,    1]

  # 相机OpenCV格式的畸变系数向量
  CvDistortions: !!opencv-matrix
    rows: 5
    cols: 1
    dt: d
    data: [-0.07177933774,    -0.5244330372,    0.003119619805,    0.0004172545196,    4.975810662]

# -----     相机的硬件参数     -----
HardwareParams:
  # 相机是否自动白平衡(布尔型)；自动白平衡：1，手动白平衡：0
  IsBalanceWhiteAuto: 1
  # 相机是否自动曝光(布尔型)；自动曝光：1，手动曝光：0
  IsExposureAuto: 0
  # 相机的曝光时间(浮点型)
  ExposureTime: 1000.0
  # 相机的增益值(浮点型)
  GainRaw: 1.0
  # 相机的伽马值(浮点型)
  Gamma: 0.6
  # 相机的亮度(整型)
  Brightness: 40